src/interfaces.rs

```diff
@@ -211,14 +222,24 @@ impl Interfaces {
     }
 
     /// Iter with removed interface
-    pub(crate) fn iter_with_removed<'a>(
+    pub(crate) fn iter_without_removed<'a>(
         &'a self,
         removed: &'a Interface,
     ) -> impl Iterator<Item = &'a Interface> + Clone {
         self.interfaces
             .values()
             .filter(|i| i.interface_name() != removed.interface_name())
     }
+
+    /// Iter with removed interfaces
+    pub(crate) fn iter_without_removed_many<'a>(
+        &'a self,
+        removed: &'a HashMap<&'a str, &'a Interface>,
```

Comment:
Are all the lifetimes here necessary, Can you not remove the ones in the hashmap generic?